packages/ra-core/src/controller/create/CreateBase.tsx

```diff
@@ -37,11 +37,20 @@ import { ResourceContextProvider } from '../../core';
  *     </CreateBase>
  * );
  */
-export const CreateBase = <RecordType extends Omit<RaRecord, 'id'> = any>({
+export const CreateBase = <
+    RecordType extends Omit<RaRecord, 'id'> = any,
+    ResultRecordType extends RaRecord = RecordType & { id: Identifier }
+>({
     children,
     ...props
-}: CreateControllerProps<RecordType> & { children: ReactNode }) => {
-    const controllerProps = useCreateController<RecordType>(props);
+}: CreateControllerProps<RecordType, Error, ResultRecordType> & {
+    children: ReactNode;
+}) => {
+    const controllerProps = useCreateController<
+        RecordType,
+        Error,
```

Comment:
why not set this to `unknown` instead of `Error`?

Reply:
Ideally we should accept an Error generic parameter in CreateBase too. I figured it would be more convenient to have a default Error